lib/puppet/daemon.rb

```diff
@@ -157,14 +157,19 @@ def remove_pidfile
 
   # Loop forever running events - or, at least, until we exit.
   def run_event_loop
-    agent_run = Puppet::Scheduler.create_job(Puppet[:runinterval], Puppet[:splay], Puppet[:splaylimit]) do
+    splaylimit = Puppet[:splay] ? Puppet[:splaylimit] : 0
+
+    agent_run = Puppet::Scheduler.create_job(Puppet[:runinterval], true, splaylimit) do |job|
+      Puppet.info "Running agent #{job}"
+
       # Splay for the daemon is handled in the scheduler
       agent.run(:splay => false)
     end
 
     reparse_run = Puppet::Scheduler.create_job(Puppet[:filetimeout]) do
       Puppet.settings.reparse_config_files
       agent_run.run_interval = Puppet[:runinterval]
+      agent_run.splay_limit = Puppet[:splay] ? Puppet[:splaylimit] : 0
```

Comment:
`splaylimit` defaults to `runinterval`. If the latter is increased, but the former isn't changed, then we don't recalculate splay. We should so that the new splay slot is chosen over the increased `runinterval` (to decrease thundering herd).

If the latter is decreased, then we should also recalculate so that our old splay slot doesn't overshoot the new `runinterval`.

Also I'd expect `splaylimit` to never be greater than `runinterval` but we don't enforce that anywhere.

Reply:
> splaylimit defaults to runinterval

This case works as expected. Changes to runinterval cause splay to be recaclulated as expected

However, if splaylimit is set explicitly, then changes to runinterval don't cause splay to be recalculated.